src/sonic-host-services/scripts/determine-reboot-cause

```diff
@@ -98,43 +99,28 @@ def find_proc_cmdline_reboot_cause():
 
     return proc_cmdline_reboot_cause
 
-def get_reboot_cause_from_platform():
-    # Until all platform vendors have provided sonic_platform packages,
-    # if there is no sonic_platform package installed, we only provide
-    # software-related reboot causes.
+
+def find_hardware_reboot_cause():
+    # Find hardware reboot cause using sonic_platform library
     try:
         import sonic_platform
         platform  = sonic_platform.platform.Platform()
         chassis  = platform.get_chassis()
-        return chassis.get_reboot_cause()
-    except ImportError as err:
+        hardware_reboot_cause_major, hardware_reboot_cause_minor = chassis.get_reboot_cause()
+        sonic_logger.log_info("Platform api returns reboot cause {}, {}".format(hardware_reboot_cause_major, hardware_reboot_cause_minor))
+        if hardware_reboot_cause_major == REBOOT_CAUSE_HARDWARE_OTHER:
+            hardware_reboot_cause_major = "{} ({})".format(hardware_reboot_cause_major, hardware_reboot_cause_minor)
+    except ImportError:
         sonic_logger.log_warning("sonic_platform package not installed. Unable to detect hardware reboot causes.")
+        hardware_reboot_cause_major, hardware_reboot_cause_minor = REBOOT_CAUSE_NON_HARDWARE, "N/A"
 
-
-def find_hardware_reboot_cause():
-    hardware_reboot_cause = None
-
-    REBOOT_CAUSE_HARDWARE_OTHER = "Hardware - Other"
-    REBOOT_CAUSE_NON_HARDWARE = "Non-Hardware"
-
-    hardware_reboot_cause_major, hardware_reboot_cause_minor = get_reboot_cause_from_platform()
-    sonic_logger.log_info("Platform api returns reboot cause {}, {}".format(hardware_reboot_cause_major, hardware_reboot_cause_minor))
-
-    if hardware_reboot_cause_major == REBOOT_CAUSE_NON_HARDWARE:
-        # The reboot was not caused by hardware. If there is a REBOOT_CAUSE_FILE, it will
-        # contain any software-related reboot info. We will use it as the previous cause.
-        pass
-    elif hardware_reboot_cause_major == REBOOT_CAUSE_HARDWARE_OTHER:
-        hardware_reboot_cause = "{} ({})".format(hardware_reboot_cause_major, hardware_reboot_cause_minor)
-    else:
-        hardware_reboot_cause = hardware_reboot_cause_major
-
-    if hardware_reboot_cause:
-        sonic_logger.log_info("Platform api indicates reboot cause {}".format(hardware_reboot_cause))
+    if hardware_reboot_cause_major:
+        sonic_logger.log_info("Platform api indicates reboot cause {}".format(hardware_reboot_cause_major))
     else:
         sonic_logger.log_info("No reboot cause found from platform api")
 
-    return hardware_reboot_cause, hardware_reboot_cause_minor
+    return hardware_reboot_cause_major, hardware_reboot_cause_minor
```

Comment:
I'm not sure why this function returns a tuple. On line 112 above, if there is a "minor" cause, we append it to the major cause, therefore the minor cause is already included in the major cause, so this seems redundant.

@sujinmkang: Can you explain why we do this now?

Reply:
@jleveque you're right. but I was expected the minor cause to be used for additional info from hardware platform vendor later. But it is currently only for the reboot_cause_hardware_other case.